geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -14,29 +15,91 @@ app.ShowLayerinfo;
 
 
 /**
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
  * @param {angular.$http} $http Angular $http service
  * @param {angular.$sce} $sce Angular $sce service
  * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
  * @ngInject
  */
-app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+app.showLayerinfoFactory = function(gettextCatalog,
+                                    $http,
+                                    $sce,
+                                    ngeoCreatePopup) {
 
   /**
    * @type {ngeo.Popup}
    */
   var popup = ngeoCreatePopup();
+  /**
+   * @type {Object.<!string, Object.<string, !angular.$q.Promise>>}
+   * @private
+   */
+  var promises_ = {};
 
   return (
-
       /**
        * @param {ol.layer.Layer} layer The layer
        */
       function(layer) {
         var title = /** @type {string} */ (layer.get('label'));
+        var localMetadata = /** @type {Object.<string>} */
+            (layer.get('metadata'));
+        var metadataUid = /** @type {string} */
+            (localMetadata['metadata_id']);
         popup.setTitle(title);
-        var content = $sce.trustAsHtml("<b>some</b> content");
-        popup.setContent(content);
-        popup.show();
+        var currentLanguage = /** @type {string} */
+            (gettextCatalog.currentLanguage);
+
+        if (!(currentLanguage in promises_)) {
+          promises_[currentLanguage] = {};
+        }
+
+        if (!(metadataUid in promises_[currentLanguage])) {
+          promises_[currentLanguage][metadataUid] = $http.jsonp(
+              'http://shop.geoportail.lu/Portail/inspire/webservices/getMD.jsp',
+              {params: {
+                'uid': metadataUid,
+                'lang': currentLanguage,
+                'cb': 'JSON_CALLBACK'
+              }}).then(
+                  angular.bind(this, function(resp) {
+                    var remoteMetadata = resp.data.root[0];
```

Comment:
If possible I'd rather work with separate objects rather writing to the service response objects.